### PR TITLE
feat: improve aggsender logs

### PR DESCRIPTION
**Pull request by joanestebanr**
## Description

Minor changes on logs og aggsender

### CODE REVIEW

File: aggsender/aggsender.go [hunk 1]
```diff
@@ -488,46 +488,52 @@ func (a *AggSender) signCertificate(certificate *agglayer.Certificate) (*agglaye
...
 
 			if err := a.storage.UpdateCertificateStatus(ctx, *certificate); err != nil {
 				err = fmt.Errorf("error updating certificate %s status in storage: %w", certificateHeader.String(), err)
 				a.log.Error(err)
-				return true, err
+				return true
 			}
 		}
+		if slices.Contains(nonSettledStatuses, certificateHeader.Status) {
+			logErrMsg += fmt.Sprintf("certificate %s is still pending, elapsed_time:%s ",
```
[the remark this answers is not in the document]
Comment (joanestebanr) [reply aggsender/aggsender.go]: Yes, for sure

File: aggsender/aggsender.go [hunk 2]
```diff
@@ -488,46 +488,49 @@ func (a *AggSender) signCertificate(certificate *agglayer.Certificate) (*agglaye
...
+		a.log.Debugf("aggLayerClient.GetCertificateHeader status [%s] of certificate %s  elapsed_time:%s",
 			certificateHeader.Status,
-			certificateHeader.String())
+			certificateHeader.String(),
+			elapsedTime)
 
 		if certificateHeader.Status != certificate.Status {
-			a.log.Infof("certificate %s changed status from [%s] to [%s]",
-				certificateHeader.String(), certificate.Status, certificateHeader.Status)
+			a.log.Infof("certificate %s changed status from [%s] to [%s] elapsed_time: %s",
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender.go:515]:
```suggestion
			a.log.Infof("certificate %s changed status from [%s] to [%s] elapsed time: %s",
```

File: aggsender/aggsender.go [hunk 3]
```diff
@@ -488,46 +488,49 @@ func (a *AggSender) signCertificate(certificate *agglayer.Certificate) (*agglaye
...
 
 			if err := a.storage.UpdateCertificateStatus(ctx, *certificate); err != nil {
 				err = fmt.Errorf("error updating certificate %s status in storage: %w", certificateHeader.String(), err)
 				a.log.Error(err)
-				return true, err
+				return true
 			}
 		}
+		if slices.Contains(nonSettledStatuses, certificateHeader.Status) {
+			a.log.Infof("certificate %s is still pending, elapsed_time:%s ",
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender.go:528]:
```suggestion
			a.log.Infof("certificate %s is still pending, elapsed time:%s ",
```

File: aggsender/aggsender_test.go [hunk 4]
```diff
@@ -280,6 +280,70 @@ func TestGetBridgeExits(t *testing.T) {
 	}
 }
 
+func TestAggSenderStart(t *testing.T) {
+	AggLayerMock := agglayer.NewAgglayerClientMock(t)
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender_test.go:284]:
```suggestion
	aggLayerMock := agglayer.NewAgglayerClientMock(t)
```

File: aggsender/aggsender_test.go [hunk 5]
```diff
@@ -280,6 +280,70 @@ func TestGetBridgeExits(t *testing.T) {
...
+	require.NotNil(t, aggSender)
+	ch := make(chan aggsendertypes.EpochEvent)
+	epochNotifierMock.EXPECT().Subscribe("aggsender").Return(ch)
+	bridgeL2SyncerMock.EXPECT().GetLastProcessedBlock(mock.Anything).Return(uint64(0), nil)
+
+	go aggSender.Start(ctx)
+	ch <- aggsendertypes.EpochEvent{
+		Epoch: 1,
+	}
+	time.Sleep(200 * time.Millisecond)
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender_test.go:309]: Why do we need this sleep at the end of the test?

**Review by goran-ethernal — APPROVED** [review]

File: aggsender/aggsender_test.go [hunk 6]
```diff
@@ -280,6 +280,70 @@ func TestGetBridgeExits(t *testing.T) {
...
+}
+
+func TestAggSenderSendCertificates(t *testing.T) {
+	AggLayerMock := agglayer.NewAgglayerClientMock(t)
```
Comment (Stefan-Ethernal) [inline aggsender/aggsender_test.go:313]:
```suggestion
	aggLayerMock := agglayer.NewAgglayerClientMock(t)
```

File: aggsender/types/epoch_notifier.go [hunk 7]
```diff
@@ -23,3 +23,6 @@ type EpochNotifier interface {
 	Start(ctx context.Context)
 	String() string
 }
+
+type BridgeL2Syncer interface {
```
Comment (Stefan-Ethernal) [inline aggsender/types/epoch_notifier.go:27]: Empty interface, strange... Can we expose some function to it?

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM, aside from minor comments

File: aggsender/types/epoch_notifier.go (hunk 7 above)
Comment (joanestebanr) [reply aggsender/types/epoch_notifier.go:27]: Whoops, that slipped through! good catch!

File: aggsender/aggsender_test.go (hunk 5 above)
Comment (joanestebanr) [reply aggsender/aggsender_test.go:309]: because is a go routine and I need to left some time to execute the code before cancelling the context

**Review by vcastellm — APPROVED** [review]
LGTM

**Pull request merged**